doc/guides/dts/index.rst

```diff
@@ -346,44 +346,534 @@ Below is sample header content generated for this node, in
+expected to also have a ``clock-frequency`` property giving the frequency, and
+an additional macro is generated:
+
+.. code-block:: c
+
+   #define DT_<node>_CLOCKS_CLOCK_FREQUENCY <frequency>
+
+Macros related to SPI GPIO chip select
+--------------------------------------
+
+.. boards/arm/sensortile_box/sensortile_box.dts has a real-world example
+
+Take these devicetree nodes as an example. where the binding for
+``vendor,spi-controller`` is assumed to have ``bus: spi``, and the bindings for
+the SPI slaves are assumed to have ``on-bus: spi``:
+
+.. code-block:: none
+
+   spi {
+	compatible = "vendor,spi-controller";
+	cs-gpios = <&gpioa 1>, <&gpioa 2>;
+	spi-slave@0 {
+		compatible "vendor,foo-spi-device";
+		reg = <0>;
+	};
+	spi-slave@1 {
+		compatible "vendor,bar-spi-device";
+		reg = <1>;
+	}
+   };
+
+Here, the unit address of the SPI slaves (0 and 1) is taken as a chip select
+address, which is used as an index into ``cs-gpios`` (a ``phandle-array``).
+``spi-slave@0`` is matched to ``<&gpioa 1>``, and ``spi-slave@1`` to
+``<&gpiob 2>``.
+
+The output for ``spi-slave@0`` and ``spi-slave@1`` is the same as if the
+devicetree had looked like this:
+
+.. code-block:: none
+
+   spi {
+	compatible = "vendor,spi-controller";
+	spi-slave@0 {
+		compatible "vendor,foo-spi-device";
+		reg = <0>;
+		cs-gpios = <&gpioa 1>;
+	};
+	spi-slave@1 {
+		compatible "vendor,bar-spi-device";
+		reg = <1>;
+		cs-gpios = <&gpioa 2>;
+	}
+   };
+
+See the ``phandle-array`` section in :ref:`dt-property-macros` for more
+information.
+
+For example, if the ``&gpioa`` node has ``label = "GPIOA"`` and 1 and 2 are pin
+numbers, then macros like the following will get generated for ``spi-slave@0``:
+
+.. code-block:: none
+
+   #define DT_<node>_CS_GPIOS_CONTROLLER    "GPIOA"
+   #define DT_<node>_CS_GPIOS_PIN           1
+
+Compatible-string and node existence macros
+-------------------------------------------
+
+An existence flag is written for each compatible strings that appears on some
+enabled node:
+
+.. code-block:: none
+
+   #define DT_COMPAT_<compatible> 1
+
+An existence flag is also written for all enabled nodes that matched some
+binding:
+
+.. code-block:: none
+
+   #define DT_INST_<instance-no.>_<compatible>    1
+
+For the example ``sim@40047000`` node above, assuming the node is the first
+node with ``compatible = "nxp,kinetis-sim"``, the macros look like this:
+
+.. code-block:: c
+
+   #define DT_COMPAT_NXP_KINETIS_SIM    1
+   #define DT_INST_0_NXP_KINETIS_SIM    1
+
+Bus-related macros
+------------------
+
+These macros get generated for nodes that appear on buses (have ``on-bus:`` in
+their binding):
+
+.. code-block:: none
+
+   #define DT_<node>_BUS_NAME                "<bus-label>"
+   #define DT_<compatible>_BUS_<bus-name>    1
+
+``<bus-label>`` is taken from the ``label`` property on the bus node, which
+must exist. ``<bus-name>`` is the identifier for the bus as given in
+``on-bus:`` in the binding.
+
+.. _reg_macros:
```

Comment:
is this wrong? `reg_macros` for flash partitions?

Reply:
Fixed, forgot to move the anchor.